### DISCUSSION TITLE
NTSC C128, 3A+, various (esp large/new) Easy Flash crts crash

### DISCUSSION THREAD

**Discussion by doughamm** [frntc/Sidekick64#58] [Q&A]
Hey!  I'm trying to run Eye of the Beholder, which works great on a BackBit Pro, on the SideKick.  I can get to the title screen but rarely further before it crashes.  I notice other Easy Flash CRTs fail as well including Zak McKracken (the one w/ mouse mode).  Is this a known issue and are there methods I can use to resolve, such as tweaking any timings?

**Comment by frntc** [comment]
I don't have an NTSC C128 myself, but have you tried the profile "RPi 3A+/3B+ (slight overclocking, choose this is C128 2MHz mode does not run reliably)" in config.txt?

**Reply by doughamm** [reply]
I did, but it doesn't seem to have improved anything.  I tried overclocking even a tad bit further but that resulted in a failure of the SK to boot.  I have some minor suspicion my pi power supply isn't up to snuff and am waiting on a new one to arrive Tuesday; I'll post back once I've ruled that out!

**Reply by doughamm** [reply]
Well, the new power supply hasn't helped with stability.  I've been working for hours at a stable set of tweaks to get Prince of Persia Easy Flash, Eye of the Beholder, and some lighter-demand carts to run on my two flat C128 NTSC machines.  The four test ROMs if anyone wishes to try are:

1. 48Khz_hifi_Elvis_Costello_[EASYFLASH].crt
2. AD&D_Pool_of_Radiance-EF-[n0].crt
3. Eye Of The Beholder.crt
4. Prince of Persia.crt (version 1.1)

With these settings I'm able to launch all titles.  Prince of Persia will inevitably crash after a couple of minutes of play though, and I just can't seem to get it stable!  EoTB will either run stable forever, or will go through spurts of crashing constantly.

sidekick64_rpi0_c128.txt:
arm_freq=1525
core_freq=600
sdram_freq=600
over_voltage=4

sidekick64.cfg:
WAIT_FOR_SIGNALS			75
WAIT_CYCLE_MULTIPLEXER		210
WAIT_CYCLE_READ				570
WAIT_CYCLE_WRITEDATA		588
WAIT_CYCLE_READ_BADLINE		504
WAIT_CYCLE_READ_VIC2		558
WAIT_CYCLE_WRITEDATA_VIC2	630
WAIT_CYCLE_MULTIPLEXER_VIC2	294
#WAIT_TRIGGER_DMA			600
#WAIT_RELEASE_DMA			600

Since I don't know what the original values are, I'm not sure how far off these tweaked values ended up being.  Assuming all tweaks in sidekick64.cfg are commented out, what exactly are the default values?

Any ideas to bring stability?  If the symptom is relative stability for a while and then a rash of crashes, could it be heat related?  2MHz NTSC is quite demanding, hence the need to overclock...

Cheers!

**Comment by frntc** [comment]
Ok, a bit of speculation here. The reason is probably the 2MHz mode, which I can't test myself (NTSC VICIIe are not common here at all).

Use the sidekick64_rpi0_c128 profile (chosen in config.txt) and modify it according to this:
-----
# RPi Zero 2: C128 profile with 1.3GHz
arm_freq=1350
over_voltage=6
core_freq=600 
sdram_freq=700
-----

At the end of sidekick64.cfg add these lines:
WAIT_FOR_SIGNALS			40
WAIT_CYCLE_MULTIPLEXER		170
WAIT_CYCLE_READ				430
WAIT_CYCLE_WRITEDATA		420
WAIT_CYCLE_READ_BADLINE		341
WAIT_CYCLE_READ_VIC2			393
WAIT_CYCLE_WRITEDATA_VIC2	449
WAIT_CYCLE_MULTIPLEXER_VIC2	226
WAIT_TRIGGER_DMA			500
WAIT_RELEASE_DMA			500

**Reply by doughamm** [reply]
Unfortunately that combo doesn't get to the main menu (LCD stuck at black).  But if it's a starting point, I'm happy to take your advice on which way to start tuning what from here!

EDIT: I lowered the overclocking to 1300/6/600/625, and it booted.  With these settings, EOB is ~90% stable...it crashed with graphical glitches anywhere from 3 minutes in to seemingly infinite (I played 15 minutes without crashing and then reset the cart to move on to other carts).  However with this setting essentially no other EF cart of any size/complexity boots, regardless of 2MHz mode or not.  

Doing further experimenting, if I crank up WAIT_CYCLE_MULTIPLEXER to 230+ the other carts boot.  Curse, Elvis Costello, etc. seem stable.  But then EOB doesn't boot.  :-/  And even in this condition POP crashes consistently after ~2 minutes of play.

I found it very interesting that lowering the arm_freq really didn't factor into the instability.  I assumed it was that, frankly, but now I'm not convinced since it seems to behave the same at 1300MHz.  However I can't get the sdram_freq above 625 without the SK failing to boot.

**Comment by doughamm** [comment]
Hi all - as with the RAD thread I started about flat NTSC C128's, I'm circling back to this one.  I've been waiting for a few weeks for my SideKick64 to return from the builder, who put it through its paces and found it to be working nominally.  (Thankfully he's in my country so cheap domestic shipping.)  Just to be safe he sent me a new one regardless.

I posted a more detailed account on forum64.de: https://www.forum64.de/index.php?thread/101221-sidekick64-aufbau-probleme-l%C3%B6sungen/&postID=1977385#post1977385

TL;DR: Working great on a plethora of C64s and a C128DCR at default settings, but still no luck getting my two flat C128's stable with it (or the RAD).  🤷

**Comment by doughamm** [comment]
Nice to see the latest D variant of the firmware supports PoP and EotB on my flat NTSC 128! Ultima V EF still seems to crash but this is a big improvement!

P.S. Does the SK64 support EF saves?

**Comment by frntc** [comment]
when/how does Ultima crash? Maybe it helps playing a bit with the timings...

EF save: yes, if saves the updated CRT when you go back to the menu

**Reply by doughamm** [reply]
Ultima V (EasyFlash version) crashes pretty much randomly after displaying the initial menu to manage savegames, journey onward, etc.  Sometimes you get to the Origin logo, but usually not.

Interesting that EF saves work; the only one I've tried is EotB, and it doesn't seem to save under any condition (including PAL C64, etc.).  Perhaps it's doing its save in an unexpected way?  This one ought to be easy to catch in action -- just run it, quick start it, walk a step away and save to flash.  Reload however, and you'll see you're taken back to the start.  (If you're not, then I have something weird on my end!)

**Reply by frntc** [reply]
what you describe is exactly how I tested it. will give it another try, but I'm pretty sure that it worked :-)

**Reply by doughamm** [reply]
Thanks! Any update on saving?
